server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -309,20 +311,65 @@ object Chapter {
                     ")",
             )
 
+        if (!serverConfig.autoDownloadNewChapters.value) {
+            log.debug { "automatic download is not configured" }
+            return
+        }
+
+        // Only download if there are new chapters, or if this is the first fetch
         val newNumberOfChapters = updatedChapterList.size
         val numberOfNewChapters = newNumberOfChapters - prevNumberOfChapters
 
         val areNewChaptersAvailable = numberOfNewChapters > 0
         val wasInitialFetch = prevNumberOfChapters == 0
 
-        // make sure to ignore initial fetch
-        val isDownloadPossible =
-            serverConfig.autoDownloadNewChapters.value && areNewChaptersAvailable && !wasInitialFetch
-        if (!isDownloadPossible) {
-            log.debug { "download is not allowed/possible" }
+        if (!areNewChaptersAvailable) {
+            log.debug { "no new chapters available" }
+            return
+        }
+
+        if (wasInitialFetch) {
+            log.debug { "skipping download on initial fetch" }
             return
         }
 
+        // Verify the manga is configured to be downloaded based on it's categories.
+        var mangaCategories = CategoryManga.getMangaCategories(mangaId)
+        // if the manga has no categories, skip checking the categories configuration for automatic
+        // download
+
+        if (mangaCategories.isEmpty()) {
+            val defaultCategory = Category.getCategoryById(Category.DEFAULT_CATEGORY_ID)
+            if (defaultCategory != null) {
+                mangaCategories = listOf(defaultCategory)
+            } else {
+                log.warn { "missing default category" }
+            }
+        }
+
+        if (mangaCategories.isNotEmpty()) {
+            var downloadCategoriesMap = Category.getCategoryList().groupBy { it.includeInDownload }
+            val unsetCategories = downloadCategoriesMap[IncludeOrExclude.UNSET].orEmpty()
+            // We only download if it's in the include list, and not in the include list.
+            // use the unset categories as the included categories if the included categories is
+            // empty
+            val includedCategories = downloadCategoriesMap[IncludeOrExclude.INCLUDE].orEmpty().ifEmpty { unsetCategories }
+            val excludedCategories = downloadCategoriesMap[IncludeOrExclude.EXCLUDE].orEmpty()
+            // This set contains all of the categories that should be downloaded.
+            val categoriesToDownload = includedCategories.subtract(excludedCategories)
+            // Now look to see if the manga has any categories in the categoriesToDownload, if so,
+            // then this manga should have it's chapters downloaded
+            val mangaDownloadCategories = mangaCategories.toSet().intersect(categoriesToDownload)
+            if (mangaDownloadCategories.isNotEmpty()) {
```

Comment:
we should only download a manga if it is not in an excluded category, even if it is in an included one

Reply:
That's how it's implemented, line 359 I subtract the excluded categories from the included categories.

Reply:
thats not how it works though, categories wont be both included and excluded, so they can't be subtracted

Reply:
Ahh right.

Reply:
Alright, updated.

Reply:
I think its still wrong, if the manga has *any* excluded category, it should not be automatically downloaded

Reply:
Oh, right, so I probably need to subtract first, before intersecting

Reply:
Just do a `if (excludedCategories.isNotEmpty()) return

Reply:
`excludedCategories` is not the _manga_ excluded categories here, it's the full set of excluded categories, so to do that I have to refactor things a bit. Let me test it.

Reply:
So it would be something like `if (excludedCategories.any { it.id in mangaCategories }) return` wouldn't it?

Reply:
Yeah that looks right, I was going to use `intersect` here too but your approach is a bit simpler

Reply:
Ok, I added some logic to check if the manga's categories is in the list of excluded categories, and it logs them if there were any before returning. I tested with a manga in two categories, one include, one exclude and got the expected behavior.